RawSpeed/LJpegDecompressor.cpp

```diff
@@ -463,27 +429,25 @@ void LJpegDecompressor::createHuffmanTable(HuffmanTable *htbl) {
  ************************************/
 
 void LJpegDecompressor::createBigTable(HuffmanTable *htbl) {
-  const uint32 bits = 14;      // HuffDecode functions must be changed, if this is modified.
+  const uint32 bits = HuffmanTable::TableBitDepth;      // HuffDecode functions must be changed, if this is modified.
   const uint32 size = 1 << bits;
-  int rv = 0;
-  int temp;
-  uint32 l;
 
   if (!htbl->bigTable)
     htbl->bigTable = (int*)_aligned_malloc(size * sizeof(int), 16);
```

Comment:
> Though i'm mostly speaking about pointer overflows that we were having in darktable. There was issues with overflow/etc, which were fixed by casting to size_t first/placing size_t-sized variable first/etc.

Intersting. I was about to ask you for some specific (real world) problem you are trying to prevent here, because I have never encountered a situation where such a reordering would solve a problem. Can you point me to a commit from darktable that did fix a real issue?